rust/flatbuffers/src/vector.rs

```diff
@@ -28,6 +28,14 @@ use primitives::*;
 #[derive(Debug)]
 pub struct Vector<'a, T: 'a>(&'a [u8], usize, PhantomData<T>);
 
+impl<'a, T: 'a> Clone for Vector<'a, T> {
+    fn clone(&self) -> Self {
+        Vector(self.0, self.1, self.2)
```

Comment:
You can use the `Copy` instance to implement the `Clone` instance. This makes it more clear that `Copy == Clone` for this type.

Specifically you can replace the body with `*self`.

However it might be late enough in the process that this should be changed in a later commit.